js/runner.go

```diff
@@ -765,6 +766,16 @@ func (u *ActiveVU) RunOnce() error {
 	ctx, cancel := context.WithCancel(u.RunContext)
 	defer cancel()
 	u.moduleVUImpl.ctx = ctx
+
+	eventIterData := event.IterData{
+		Iteration:    uint64(u.iteration),
+		VUID:         u.ID,
+		ScenarioName: u.scenarioName,
+	}
+	if u.Runner.preInitState.Events != nil {
```

Comment:
> The biggest change seems to be to execution package where no such helper exists and there are >10 uses.

Not really. It's used much more in `js` tests:

```
$ rg -g '*_test.go' -c 'TestPreInitState\{'
execution/scheduler_int_test.go:1
execution/scheduler_ext_exec_test.go:4
execution/scheduler_ext_test.go:3
cmd/runtime_options_test.go:2
converter/har/converter_test.go:1
js/init_and_modules_test.go:2
js/modules/k6/marshalling_test.go:1
js/runner_test.go:28
js/console_test.go:1
js/module_loading_test.go:11
js/bundle_test.go:1
js/modules/k6/html/html_test.go:1
js/share_test.go:1
js/modules/k6/metrics/metrics_test.go:3
lib/executor/common_test.go:1
api/v1/group_routes_test.go:1
metrics/engine/ingester_test.go:1
metrics/engine/engine_test.go:3
```

Many of them pass it to `NewFromArchive`, which in some cases doesn't matter if they don't run `VU.RunOnce()`, but most do. Since they check the behavior of creating the archive from the first runner, calling `getSimpleRunner()` wouldn't help in these cases.

We could create a separate helper that all these tests call, and just do the change there, but... it's a lot of tedious work. :sweat: I started doing something like that, but gave up and just added the nil check, hoping we could settle on a better solution. If we can't think of anything, I'll consider refactoring the tests... soon-ish.

> Also FYI you should be able to do this check in the Emit call before you use the nil instance of the Events.

I guess so, but that's just moving it to a less relevant place. I'd rather see it here where we know it can be nil, then in `Emit` where it mostly shouldn't be.

Reply:
I removed the nil checks in f7588d616, since this now uses the local per-VU event system, which is guaranteed to be defined here.